### PR TITLE
[GH-712] Move RequireJS deps to package.json and fix build

### PR DESCRIPTION
**Pull request by buunguyen**
This avoids RequireJS packages to be pulled to user apps. Fixes config for build to work. Also adds hpcc-viz-form.js. 

@GordonSmith please review.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,27 +3,29 @@
...
-    "csso": "^1.3.11",
-    "gulp-git": "^1.2.3",
-    "gulp-bump": "^0.3.0",
-    "gulp-filter": "^2.0.2",
-    "gulp-tag-version": "^1.2.1",
-    "yargs": "^3.8.0",
-    "jshint-stylish": "~1.0.2",
-    "gulp-jshint": "~1.10.0",
-    "gulp-jscs": "~1.6.0"
+    "requirejs-plugins": "^1.0.2",
```
Comment (GordonSmith) [inline package.json]: This needs to use my branch, otherwise the visualizations will fail to load in DOJO AMD environments.

**Comment by GordonSmith** [comment]
Can you rebase this for master (which will be 1.6.0) shortly - mainly because there are a bunch of additional tests which make catch some "post build" issues.

**Comment by buunguyen** [comment]
@GordonSmith will do. Is there any chance we can make this fix land on 1.4.4? It will help us get rid of a bower exclusion workaround we have to add in all projects. In fact, with all the fixes so far for 1.4.4, I hope we'll get rid of all workarounds.

**Comment by GordonSmith** [comment]
I suspect 1.4.4 and 1.6.0 will be released around the same time.  If you want it in 1.4.4 I would also suggest not sorting the requires (as its contributing to the upmerge clash)

**Comment by buunguyen** [comment]
@GordonSmith npm does the sorting itself, probably a new feature. But I can certainly rearrange it manually.

I didn't know 1.6.0 and 1.4.4 can be released on the same day. What's the difference between them, as I need to use one of them. As long as the non-AMD fix + other bugs/prs I submitted go into one of these version, I'll use it. So far I've been thinking they go to 1.4.4, not sure if these commits will be rebased to 1.6.0?

**Comment by GordonSmith** [comment]
1.4.4 should be 100% bug fixes to 1.4.2, 1.6.0 will include new features (which I consider this to be) (a 2.x.x. release would signify breaking changes)

**Comment by buunguyen** [comment]
@GordonSmith got you. Can I assume all fixes in 1.4.4 will also go to 1.6.0?

**Comment by GordonSmith** [comment]
@buunguyen Yes

**Pull request closed without merging**